models/contacts.js

```diff
@@ -1,19 +1,87 @@
-// const fs = require('fs/promises')
+const fs = require("fs/promises");
+const path = require("path");
 
-const listContacts = async () => {}
+const contactsPath = path.join(__dirname, "/contacts.json");
 
-const getContactById = async (contactId) => {}
+const listContacts = async () => {
+	try {
+		const data = await fs.readFile(contactsPath);
+		return data.toString();
+	} catch (err) {
+		console.log(err.message);
+	}
+};
 
-const removeContact = async (contactId) => {}
+const getContactById = async (contactId) => {
+	try {
+		const data = await fs.readFile(contactsPath);
+		const parsedData = JSON.parse(data);
+		const contact = parsedData.filter((el) => el.id === contactId);
+		return contact;
+	} catch (err) {
+		console.log(err.message);
+	}
+};
 
-const addContact = async (body) => {}
+const removeContact = async (contactId) => {
+	const contactList = await fs.readFile(contactsPath);
+	const parsedContactList = JSON.parse(contactList);
+	const contactIndex = parsedContactList.findIndex(
+		(contact) => contact.id === contactId
+	); // Find the index of the contact with the given ID
+	if (contactIndex !== -1) {
+		parsedContactList.splice(contactIndex, 1); // Remove the contact from the array
+		await fs.writeFile(
+			contactsPath,
+			JSON.stringify(parsedContactList, null, 2),
+			"utf8"
+		);
+		return { message: "contact deleted" };
+	} else {
+		return { message: "Not found" };
+	}
+};
 
-const updateContact = async (contactId, body) => {}
+const addContact = async (contact) => {
+	const contactList = await fs.readFile(contactsPath);
+	const parsedContactList = JSON.parse(contactList);
+	parsedContactList.push(contact);
+	await fs.writeFile(
```

Comment:
Actually, this also can be re-used. 
What do you think about creating method `storeContacts` with `contacts` parameter.
Body of this function can be 
```
await fs.writeFile(
		contactsPath,
		JSON.stringify(contacts, null, 2),
		"utf8"
```